test/unit/default_hostgroup_test.rb

```diff
@@ -21,10 +21,14 @@ def setup_hostgroup_matchers
     FactoryGirl.create(:setting,
                        name: 'force_host_environment',
                        category: 'Setting::DefaultHostgroup')
+    FactoryGirl.create(:setting,
+                       name: 'use_and_operator',
+                       category: 'Setting::DefaultHostgroup')
     # Set the defaults
     Setting[:force_hostgroup_match] = false
     Setting[:force_hostgroup_match_only_new] = true
     Setting[:force_host_environment] = true
+    Setting[:use_and_operator] = false
```

Comment:
This will ensure the "OR" code is still tested, but are you able to add tests for the "AND" case?